pkg/apis/core/validation/validation_test.go

```diff
@@ -18660,6 +18733,335 @@ func testAnyDataSource(t *testing.T, ds, dsRef bool) {
+				{
+					Name: "foo",
+					VolumeSource: core.VolumeSource{
+						Ephemeral: &core.EphemeralVolumeSource{
+							VolumeClaimTemplate: &core.PersistentVolumeClaimTemplate{
+								Spec: core.PersistentVolumeClaimSpec{
+									AccessModes: []core.PersistentVolumeAccessMode{
+										core.ReadWriteOnce,
+										core.ReadOnlyMany,
+									},
+									Resources: core.ResourceRequirements{
+										Requests: core.ResourceList{
+											core.ResourceName(core.ResourceStorage): resource.MustParse("10G"),
+										},
+										Limits: core.ResourceList{
+											core.ResourceName(core.ResourceStorage): resource.MustParse("20G"),
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			expectedFail: false,
+		},
+		{
+			testName: "a volume deleted and another updated with limits, should fail",
+			volumes: []core.Volume{
+				{
+					Name: "foo",
+					VolumeSource: core.VolumeSource{
+						Ephemeral: &core.EphemeralVolumeSource{
+							VolumeClaimTemplate: &core.PersistentVolumeClaimTemplate{
+								Spec: core.PersistentVolumeClaimSpec{
+									AccessModes: []core.PersistentVolumeAccessMode{
+										core.ReadWriteOnce,
+										core.ReadOnlyMany,
+									},
+									Resources: core.ResourceRequirements{
+										Requests: core.ResourceList{
+											core.ResourceName(core.ResourceStorage): resource.MustParse("10G"),
+										},
+										Limits: core.ResourceList{
+											core.ResourceName(core.ResourceStorage): resource.MustParse("30G"),
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			oldVolumes: []core.Volume{
+				{
+					Name: "foo",
+					VolumeSource: core.VolumeSource{
+						Ephemeral: &core.EphemeralVolumeSource{
+							VolumeClaimTemplate: &core.PersistentVolumeClaimTemplate{
+								Spec: core.PersistentVolumeClaimSpec{
+									AccessModes: []core.PersistentVolumeAccessMode{
+										core.ReadWriteOnce,
+										core.ReadOnlyMany,
+									},
+									Resources: core.ResourceRequirements{
+										Requests: core.ResourceList{
+											core.ResourceName(core.ResourceStorage): resource.MustParse("10G"),
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+				{
+					Name: "bar",
+					VolumeSource: core.VolumeSource{
+						Ephemeral: &core.EphemeralVolumeSource{
+							VolumeClaimTemplate: &core.PersistentVolumeClaimTemplate{
+								Spec: core.PersistentVolumeClaimSpec{
+									AccessModes: []core.PersistentVolumeAccessMode{
+										core.ReadWriteOnce,
+										core.ReadOnlyMany,
+									},
+									Resources: core.ResourceRequirements{
+										Requests: core.ResourceList{
+											core.ResourceName(core.ResourceStorage): resource.MustParse("100G"),
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			expectedFail: true,
+		},
+		{
+			testName: "a volume deleted and another added with limits, should succeed",
```

Comment:
Should this pass? I thought we wanted to only pass validation with limits set, if an existing resource had limits. Creating a new volume with limits should fail (or have I lost track of something?)

Reply:
Yes, it should paas. as `ValidateVolumesUpdate` only validates for updates, validation for creating should be performed in `ValidateXXXSpec` as mentioned in the above comments response.

Reply:
What about a pod whose old volumes are [{foo, limit: 10G}, {bar, <no limit>}], and we update this to [{foo, limit: 10G}, {bar, limit: 10G}], or even just [{baz, limit:10G}]. All of those should fail, right? I'm not sure that's true in your PR.

Reply:
You are right, all of these should fail. 

Thanks for pointing that, I have fixed it, and two test cases: `"one of two old volumes has limits set, update as both have limits set, should failed"` and `one of two old volumes has limits set, update as another volume has limits set, should failed`, have been added accordingly to verify it.